extensions/standard-processors/processors/AttributesToJSON.cpp

```diff
@@ -0,0 +1,174 @@
+  std::string value;
+  if (context->getProperty(AttributesList.getName(), value) && !value.empty()) {
+    attribute_list_ = utils::StringUtils::splitAndTrimRemovingEmpty(value, ",");
+  }
+  if (context->getProperty(AttributesRegularExpression.getName(), value) && !value.empty()) {
+    attributes_regular_expression_ = std::regex(value);
+  }
+  write_destination_ = WriteDestination::parse(utils::parsePropertyWithAllowableValuesOrThrow(*context, Destination.getName(), WriteDestination::values()).c_str());
+  context->getProperty(IncludeCoreAttributes.getName(), include_core_attributes_);
+  context->getProperty(NullValue.getName(), null_value_);
+}
+
+bool AttributesToJSON::isCoreAttributeToBeFiltered(const std::string& attribute) const {
+  return !include_core_attributes_ && core_attributes_.find(attribute) != core_attributes_.end();
+}
+
+std::unordered_set<std::string> AttributesToJSON::getAttributesToBeWritten(const std::map<std::string, std::string>& flowfile_attributes) const {
+  std::unordered_set<std::string> attributes;
+
+  for (const auto& attribute : attribute_list_) {
+    if (!isCoreAttributeToBeFiltered(attribute)) {
+      attributes.insert(attribute);
+    }
+  }
+
+  if (attributes_regular_expression_) {
+    for (const auto& [key, value] : flowfile_attributes) {
+      if (!isCoreAttributeToBeFiltered(key) && std::regex_match(key, attributes_regular_expression_.value())) {
+        attributes.insert(key);
+      }
+    }
+  }
+
+  return attributes;
+}
+
+void AttributesToJSON::addAttributeToJson(rapidjson::Document& document, const std::string& key, const std::string& value) {
+  rapidjson::Value json_key(key.c_str(), document.GetAllocator());
+  rapidjson::Value json_val;
+  if (!value.empty() || !null_value_) {
+    json_val.SetString(value.c_str(), document.GetAllocator());
+  }
+  document.AddMember(json_key, json_val, document.GetAllocator());
+}
+
+std::string AttributesToJSON::buildAttributeJsonData(std::map<std::string, std::string>&& flowfile_attributes) {
+  auto root = rapidjson::Document(rapidjson::kObjectType);
+
+  if (!attribute_list_.empty() || attributes_regular_expression_) {
+    auto attributes_to_write = getAttributesToBeWritten(flowfile_attributes);
+    for (const auto& key : attributes_to_write) {
+      addAttributeToJson(root, key, flowfile_attributes[key]);
+    }
+  } else {
+    for (const auto& [key, value] : flowfile_attributes) {
+      if (!isCoreAttributeToBeFiltered(key)) {
+        addAttributeToJson(root, key, value);
+      }
+    }
+  }
+
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  root.Accept(writer);
+  return buffer.GetString();
+}
+
+void AttributesToJSON::onTrigger(core::ProcessContext* /*context*/, core::ProcessSession* session) {
+  auto flow_file = session->get();
+  if (!flow_file) {
+    return;
+  }
+
+  auto json_data = buildAttributeJsonData(flow_file->getAttributes());
```

Comment:
as the ``attributes_` in the `FlowFile` is of type `FlatMap`, `getAttributes` constructs a new `std::map` for backwards compatibility, we should avoid this if we can IMO